### ISSUE TITLE
Memory leak in pod's messageFilter

### ISSUE THREAD

**Issue by jpcoenen** [suborbital/grav#86]
Hey!

I have been digging into a memory leak with our product that uses Grav. During my search, I noticed ever-increasing memory consumption by `github.com/suborbital/grav/grav.(*messageFilter).FilterUUID`. Some further digging, has led to the following theory:

[In `pod.Send()`](https://github.com/suborbital/grav/blob/74abd521aca88cf0d016b291a61e6b48359d498e/grav/pod.go#L91) `p.FilterUUID(msg.UUID(), false)` is called. In `FilterUUID()`, the UUID of the sent message is added to the `UUIDMap` map. This UUID seems to never be removed from the map again, leading to an increase in memory consumption every time `pod.Send()` is called.

I am happy to open an MR to address this, but my understanding of the internals of Grav are not super strong. So some guidance is certainly welcome!

**Comment by cohix** [comment]
@jpcoenen Yes you're absolutely right, that is a memory leak! I found this a while back but haven't had a chance to design a good solution. One idea I had was using a MsgBuffer (one of the other internal data structures), but haven't had time to sufficiently test. Happy to get on a call some time and pair on it, if you'd like. Ask Rick for my contact info or we can create a shared Slack channel.

**Comment by cohix** [comment]
@jpcoenen that is an excellent idea, we'd just have to add a UUID to each pod! Much simpler. Happy to pair on it if you'd like.

Yes the ring buffer has several possible edge cases which is why I wasn't super motivated to implement it here 😛

**Comment by jpcoenen** [comment]
Opened a quick MR for this: https://github.com/suborbital/grav/pull/87

Let me know if it aligns with what you had in mind. Feel free to propose any changes.
